File: Program Tabel Fungsi.py

nama = "Salsabilla Tri Putri"
nim = "2310431017"
shift = "2"
print()
print("TUGAS 5 PRAKTIKUM ADP\n")
print("Materi : ARRAY")
print("Nama   : "+nama)
print("NIM    : "+nim)
print("Shift  : "+shift)

print()
import math
while True:
    n = int(input("Banyak data : "))

    x = [0] * n
    fx = [0] * n
    gx = [0] * n
    hx = [0] * n

    for i in range(n):
        x[i] = int(input(f"Nilai x ke-{i+1} : "))
        if x[i] > 0:
            fx[i] = int(3*x[i]**2 + 7*x[i] - 2)
        elif x[i] < 0:
            fx[i] = int(2*x[i]**2 - 5*x[i] - 4)
        else:
            fx[i] = 0
        gx[i] = int(fx[i]**2 - math.sqrt(fx[i]))
        hx[i] = int(fx[i] * gx[i])

    print("\n --------------------------------------------------")
    print("|                   TABEL HASIL                    |")
    print("|--------------------------------------------------|")
    print("|    x    |    f(x)    |    g(x)    |     h(x)     |")
    print("|--------------------------------------------------|")
    for i in range(n):
        print(f"|    {x[i]}   |     {fx[i]}    |   {gx[i]}   |   {hx[i]}  |")
    print(" -------------------------------------------------- ")
    print()
    pilihan = input("Apakah Anda ingin kembali memasukkan nilai x ? (Y/N): ")
    if pilihan.upper() != 'Y':
        break

File: Program Tabel Fungsi.cpp

#include <iostream>
#include <cmath>
using namespace std;

const int MAX_DATA = 100;

int main() {
    string nama = "  Salsabilla Tri Putri";
    string nim = "  2310431017";
    string shift = "  2";

    cout << endl;
    cout << "  TUGAS 5 PRAKTIKUM ADP\n" << endl;
    cout << "  Materi : ARRAY" << endl;
    cout << "  Nama   : " << nama << endl;
    cout << "  NIM    : " << nim << endl;
    cout << "  Shift  : " << shift << endl;
    cout << endl;

    while (true) {
        int n;
        cout << "  Banyak data : ";
        cin >> n;

        if (n > MAX_DATA) {
            cout << "  Jumlah data melebihi batas maksimum." << endl;
            continue;
        }

        int x[MAX_DATA], fx[MAX_DATA], gx[MAX_DATA], hx[MAX_DATA];

        for (int i = 0; i < n; ++i) {
            cout << "  Nilai x ke-" << i+1 << " : ";
            cin >> x[i];
            if (x[i] > 0) {
                fx[i] = 3*x[i]*x[i] + 7*x[i] - 2;
            } else if (x[i] < 0) {
                fx[i] = 2*x[i]*x[i] - 5*x[i] - 4;
            } else {
                fx[i] = 0;
            }
            gx[i] = fx[i]*fx[i] - sqrt(fx[i]);
            hx[i] = fx[i] * gx[i];
        }

        cout << "\n   --------------------------------------------------" << endl;
        cout << "  |                   TABEL HASIL                    |" << endl;
        cout << "  |--------------------------------------------------|" << endl;
        cout << "  |    x    |    f(x)    |    g(x)    |     h(x)     |" << endl;
        cout << "  |--------------------------------------------------|" << endl;
        for (int i = 0; i < n; ++i) {
            cout << "  |    " << x[i] << "   |     " << fx[i] << "    |   " << gx[i] << "   |   " << hx[i] << "  |" << endl;
        }
        cout << "   -------------------------------------------------- " << endl;
        cout << endl;

        char pilihan;
        cout << "  Apakah Anda ingin kembali memasukkan nilai x ? (Y/N): ";
        cin >> pilihan;
        if (pilihan != 'Y' && pilihan != 'y') {
            break;
        }
    }
}
